### PR TITLE
Parser

### LINKED ISSUE #25: Add command line parser

### PR DESCRIPTION
**Pull request by nikklassen**
closes #25

### CODE REVIEW

File: src/parser.rs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    fn parse_trailing_ws() {
+        let word = parse_command("cmd    ").unwrap();
+        assert_vec_eq(word, vec![to_word("cmd")]);
+    }
+
+    fn to_word(s: &str) -> AST {
+        AST::Word(s.to_string())
+    }
+
+    fn assert_vec_eq<T: PartialEq + Debug>(v1: Vec<T>, v2: Vec<T>) {
```
Comment (danReynolds) [inline src/parser.rs]: When I run `cargo test` it spits out *use of undeclared trait name* `Debug` as a result of this.

Comment (nikklassen) [inline src/parser.rs]: We don't actually need that trait so I'll just remove it

**Comment by nikklassen** [comment]
It shouldn't be many because then the parser would expect parse an empty string.  But since it's specifically a "words" parser it should parse works.  If you want to accept an empty string that should be at the highest level, basically make the entire parser optional.

**Comment by danReynolds** [comment]
Okay if do that then because this change breaks immediate returns

**Comment by nikklassen** [comment]
Sure. I'll let you do that, then merge it tonight.

**Pull request merged**